core/primitives-core/src/hash.rs

```diff
@@ -116,6 +116,12 @@ impl From<&CryptoHash> for Vec<u8> {
     }
 }
 
+impl From<CryptoHash> for [u8; 32] {
+    fn from(hash: CryptoHash) -> [u8; 32] {
+        (hash.0).0
+    }
+}
```

Comment:
In contrast, this impl is 100% warranted: the conversion here matches the spirit of the From trait (lossless owned value to owned value) and we can't add `fn new` method to `[u8; 32]`.